api/src/main/java/org/openmrs/api/TagService.java

```diff
@@ -0,0 +1,156 @@
+	 * @return Tag with given internal identifier
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	Tag getTag(Integer tagId);
+	
+	/**
+	 * Returns a list of Tag objects having the given tag, or having tags containing the substring.
+	 * Partial Search supported
+	 * 
+	 * @param tag the tag (String) which the
+	 * @return List of Tag Objects
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	List<Tag> getTags(String tag);
+	
+	/**
+	 * Returns a list of All Tags.
+	 * 
+	 * @return list Of Tags (String)
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	List<String> getAllTags();
+	
+	/**
+	 * Retrieves an Object with given object uuid and object type from the Openmrs database.
+	 * 
+	 * @param objectUuid the UUID of the object
+	 * @param objectType the java class of the object (eg. Concept.class)
+	 * @return
+	 */
+	<T extends OpenmrsObject> T getObject(Class<T> objectType, String objectUuid);
+	
+	/**
+	 * It adds a tag to the openmrs object, if such a tag does not already exist on the object. A
+	 * warning will be logged if the object does not exist in the database
+	 * 
+	 * @param openmrsObject the object to which the tag has to be added
+	 * @param tag the tag (string) which is to be added
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	Tag addTag(OpenmrsObject openmrsObject, String tag);
+	
+	/**
+	 * It returns a list of tags added to the openmrs object
+	 * 
+	 * @param openmrsObject the object whose tags needs to obtained
+	 * @return a List of Tag Objects having the same object type and uuid as the Openmrs Object
+	 *         parameter
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	List<Tag> getTags(OpenmrsObject openmrsObject);
+	
+	/**
+	 * returns a List of Tag Objects which have an Object Type that belongs to the objectTypes
+	 * parameter and a tag which belongs to the tags (List<String>) parameter. returns a list of Tag
+	 * Objects which have a tag belonging to the List, if objectTypes list is empty. Throws an API
+	 * Exception if tags list is empty.
+	 * 
+	 * @param objectTypes a list of object types (objectTypes should be the Java Class Name)
+	 * @param tags a list of tags
+	 * @return a list of tag objects which have an object_type and a tag from the parameters
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	List<Tag> getTags(List<String> objectTypes, List<String> tags);
+	
+	/**
+	 * returns a list of OpenMrs Objects which have an objectType from the parameters, and have all
+	 * the tags.
+	 * 
+	 * @param objectTypes the permissible object_types to be searched against
+	 * @param tags the list of tags that all objects should have
+	 * @return a list of Openmrs Objects
+	 */
+	List<OpenmrsObject> getObjectsWithAllTags(List<String> objectTypes, List<String> tags);
+	
+	/**
+	 * Obtains the java class from a String.
+	 * 
+	 * @param className the string java class name (ex. org.openmrs.Concept)
+	 * @param <T>
+	 * @return the OpenmrsObject class
+	 */
+	public <T extends OpenmrsObject> T toClass(String className);
```

Comment:
Why is this service method? Why can't you just use OpenmrsClassLoader.getInstance().loadClass()? If you want encapsulate it in a single method, you need to create a utility class with such convenience method(s)